app/controllers/deploys_controller.rb

```diff
@@ -178,4 +184,19 @@ def find_deploy
   def active_deploy_scope
     current_project ? current_project.deploys.active : Deploy.active
   end
+
+  # Creates a CSV for @deploys as a result of the search query limited to 1000 for speed
+  def realtime_csv
+    csv_limit = (params[:limit].presence || 1000).to_i
+    CSV.generate do |csv|
+      csv << Deploy.csv_header
+      @deploys.limit(csv_limit).each { |deploy| csv << deploy.csv_line }
+      csv << ['-', 'count:', @deploys.limit(csv_limit).count]
+      csv << ['-', 'params:', params]
+      if @deploys.count > csv_limit
```

Comment:
could save that .count query (which will be expensive with lots of deploys) and ask for 1001 and then check if we got 1001 ... and always pop 1